### PR TITLE
[Do-Not-Merge] Add deprecated notice to terra-overlay

### LINKED ISSUE #3331: [terra-overlay] Deprecate terra-overlay
# Feature Request

## Description

terra-application provides an overlay and a loading overlay. Those components are recommended over usage of this component.

Update the terra-overlay package to indicate that it is deprecated.
Update terra-overlay documentation to indicate that it will be deprecated in the future using the deprecation notice.
Create a deprecation guide to direct people on  how to migrate to terra-applications overlay.

### PR DESCRIPTION
**Pull request by pranav300**
### Summary

Added a deprecated notice to terra-overlay

`The plan is to wait till terra-application v2 is released and then update this PR with correct links for migration guide.`

`Till then this can work as a template for deprecation and maintenance PR's.`

Closes #3331 

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-overlay/README.md [hunk 1]
```diff
@@ -4,9 +4,23 @@
 [![NPM version](https://badgen.net/npm/v/terra-overlay)](https://www.npmjs.org/package/terra-overlay)
 [![Build Status](https://badgen.net/travis/cerner/terra-core)](https://travis-ci.com/cerner/terra-core)
 
-The Terra Overlay component is a component that displays an overlay relative to the container that triggered the overlay. This component blocks interactions and fades out all elements of the triggering container.
+## Deprecation Notice
```
Comment (yuderekyu) [inline packages/terra-overlay/README.md:7]: if we converted this into mdx, we could use our deprecation notice component: https://engineering.cerner.com/terra-ui/dev_tools/cerner/documentation-components/notice#notices-for-implementation-suggestions

Comment (mjhenkes) [reply packages/terra-overlay/README.md:7]: We shouldn't convert the readme to mdx. It needs to be renderable by github.

Comment (mjhenkes) [reply packages/terra-overlay/README.md:7]: and npm

Comment (yuderekyu) [reply packages/terra-overlay/README.md:7]: oops good call, didn't realize this was the readme

**Comment by mjhenkes** [comment]
For any of these depreciation notices we need to do the following.

- [x] update the repo readme and badge
- [ ] update any of the component doc pages, using the [notices component](https://engineering.cerner.com/terra-ui/dev_tools/cerner/documentation-components/notice#notices-for-development-support-status). In this case, the `maintainance` notice variant would be appropriate with similar phrasing as used for [terra-application-navigation](https://engineering.cerner.com/terra-ui/components/terra-application-navigation/application-navigation/about#terra-application-navigation)
- [x] update the component readme (using just straight markdown), update the badge
- [ ] adding an independent deprecation guide doc page. This doc should contain what deprecation means, why we've made the change and how consumers should move forward. (Basically what you've already documented [here](https://github.com/cerner/terra-core/pull/3357/files#diff-3da4da81c31915d64f77355b45094435c903c91225e3b13837462e85526921dfR11-R23). I will want to make some tweaks to the phrasing here but you have a good start.

---After PR----
- [ ] npm deprecate with a message (we should use the exact same message as what is in our notices component on the main doc pages so these match, plus also on the component readme.md - all can link to our deprecation guide doc).

File: packages/terra-overlay/README.md [hunk 2]
```diff
@@ -4,9 +4,23 @@
...
 
-A Loading Overlay is a subcomponent that displays an overlay with a spinner icon and loading message.
+The terra-overlay npm package is deprecated as of 3.62.0. terra-overlay will be removed from the repo in a future version, but the package will remain on npm.
```
Comment (emilyrohrbough) [inline packages/terra-overlay/README.md:9]:
```suggestion
The terra-overlay npm package is deprecated as of `3.62.0`. terra-overlay will be removed from the repo in a future version, but the package will remain on npm.
```

File: packages/terra-overlay/README.md [hunk 3]
```diff
@@ -4,9 +4,23 @@
...
+
+### What deprecation means
+Any apps that currently use terra-overlay will continue to work as-is.
+However, the terra-overlay package will no longer continue to be supported / developed.
+
+## Why?
+Terra Overlay design is a bad pattern and hence it is being deprecated.
```
Comment (emilyrohrbough) [inline packages/terra-overlay/README.md:16]: What do you mean by this is a bad pattern? being devils advocate, without any background context, I don't really feel like this sells me on why it should deprecated.

Also what is different about `terra-application-loading-overlay` that make it a better pattern?

Comment (mjhenkes) [reply packages/terra-overlay/README.md:16]: Yeah, i was going to get to that once we got some other things cleaned up, but it's not a bad pattern it's just an old pattern and now we want people to pull from terra-application instead. There will have to be some more tweaking of the wording here.

Comment (tbiethman) [reply packages/terra-overlay/README.md:16]: I'd say the biggest reason for this deprecation is to integrate the presentation of overlays with the overall application framework to ensure accessibility. Improving the ease of use of these components is a plus, too.

Comment (pranav300) [reply packages/terra-overlay/README.md]: Updated here https://github.com/cerner/terra-core/pull/3357/commits/a63aa91b91c0854bcfb22e4a435822771f932e7f, with what @tbiethman mentioned ☝️.

**Review by dkasper-was-taken — APPROVED** [review]

**Comment by pranav300** [comment]
Closing this PR. Due to delay in terra-application v2 release, marking these components deprecated seems premature.

The deprecations of this will be done once terra-application v2 is released.

cc/ @ryanthemanuel @ShettyAkarsh

**Pull request closed without merging**